File: v1/parked_legacy/TarsEngine.FSharp.Main/Services/IMetascriptService.fs
namespace TarsEngine.FSharp.Main.Services

open System.Threading.Tasks

/// <summary>
/// Service for executing metascripts
/// </summary>
type IMetascriptService =
    /// <summary>
    /// Executes a metascript asynchronously.
    /// </summary>
    /// <param name="metascript">The metascript to execute.</param>
    /// <returns>A task that represents the asynchronous operation. The task result contains the execution result.</returns>
    abstract member ExecuteMetascriptAsync : metascript:string -> Task<obj>

File: archive/v1/parked_csharp/Legacy_CSharp_Projects/TarsEngine/Services/Interfaces/IMetascriptService.cs
namespace TarsEngine.Services.Interfaces;

/// <summary>
/// Interface for the metascript service
/// </summary>
public interface IMetascriptService
{
    /// <summary>
    /// Executes a metascript
    /// </summary>
    /// <param name="metascript">Metascript to execute</param>
    /// <returns>Result of the metascript execution</returns>
    Task<object> ExecuteMetascriptAsync(string metascript);
}
